AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -1139,7 +1139,14 @@ - (void)didMoveToWindow
   // Updating the visible node index paths only for not range managed nodes. Range managed nodes will get their
   // their update in the layout pass
   if (![node supportsRangeManagedInterfaceState]) {
-    [_rangeController visibleNodeIndexPathsDidChangeWithScrollDirection:self.scrollDirection];
+    // Grab a strong reference for data source and delegate to be sure they are not going away while executing
+    // the range update. This can happen in range updates while going back in the view controller hierarchy
+    __block id<ASCollectionDataSource> asyncDataSource = _asyncDataSource;
```

Comment:
Technically these don't need to be block variables, just referenced inside the block, although perhaps this is the most explicit way?